### PR TITLE
Add the Conditional Logic and Dynamic Texts help topic

### PR DESCRIPTION
**Pull request by RomanTsukanov**

### CODE REVIEW

File: docs/design-survey-conditional-logic.md [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+    "setValue": "adult"
+  }]
+}
+```
+
+---
+
+### `copyvalue`
+
+Copies a value from one question to another. The `fromName` property specifies the source question; the `setToName` property specifies the target question.
```
Comment (dmitry-eliseev-devexpress) [inline docs/design-survey-conditional-logic.md:660]:
```suggestion
Copies a value from between questions. The `fromName` property specifies the source question; the `setToName` property specifies the target question.
```

File: docs/design-survey-conditional-logic.md [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+    "setToName": "shippingAddress"
+  }]
+}
+```
+
+---
+
+### `runexpression`
+
+If the `expression` is `true`, runs another expression specified by the `runExpression` property. You can also save the result of `runExpression` as a question value. For this, assign the question's name to the `setToName` property.
```
Comment (dmitry-eliseev-devexpress) [inline docs/design-survey-conditional-logic.md:690]: > If the `expression` is `true`, **runs** another expression specified by the `runExpression` property.

Who runs?

File: docs/design-survey-conditional-logic.md [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+- **Boolean expression**    
+  An expression that evaluates to `true` or `false`. Boolean expressions are widely used to implement conditional logic. Refer to the following help topic for more information: [Conditional Visibility](#conditional-visibility).
+
+Expressions can include question names, variables, and calculated values (described in the [Dynamic Texts](#dynamic-texts) section). Plus expressions can use [built-in](#built-in-functions) and [custom functions](#custom-functions).
+
+### Built-In Functions
+
+Functions allow you to perform additional calculations within an expression. One expression can contain multiple function calls.
+
+Functions can accept arguments. For example, the following expression shows the built-in [`age`](#age) and [`iif`](#iif) functions. `age` accepts the value of a `birthdate` question. `iif` accepts three arguments: a condition, a value to return when the condition is truthy, and a value to return when the condition is falsy.
```
[the remark this answers is not in the document]
Comment (RomanTsukanov) [reply docs/design-survey-conditional-logic.md:169]: [Truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) and [falsy](https://developer.mozilla.org/en-US/docs/Glossary/falsy) are legit terms in JavaScript world.

File: docs/design-survey-conditional-logic.md (hunk 2 above)
Comment (RomanTsukanov) [reply docs/design-survey-conditional-logic.md:690]: The trigger obviously. This is the same pattern used in the previous descriptions of triggers, just with an if-part.

File: docs/design-survey-conditional-logic.md [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+
+### Calculated Values
+
+Calculated values allow you to register an [expression](#expressions) under a required name. If the expression includes [questions](#question-values), [variables](#variables), or [functions](#built-in-functions), it is recalculated each time their values are changed.
+
+To configure a calculated value, define the [`calculatedValues`](https://surveyjs.io/Documentation/Library?id=surveymodel#calculatedValues) array in the survey JSON definition. Each object in this array should contain the following fields:
+
+- `name` - A name that identifies the calculated value.
+- `expression` - An expression that returns the calculated value.
+- `includeIntoResult` - A Boolean property that specifies whether to include the calculated value into survey results.
```
Comment (annnke) [inline docs/design-survey-conditional-logic.md:115]: `includeIntoResult` - A Boolean property that specifies whether to include the calculated value in survey results.

File: docs/design-survey-conditional-logic.md [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
+  // Preserve the context variable to access the `returnResult` callback later
+  var self = this;
+  setTimeout(function() {
+    // Return the function result via the callback
+    self.returnResult(yourValue);
+  }, 100);
+}
+```
+
+After you implement an asynchronous function, register it in the `FunctionFactory`. The third parameter specifies if this function is asynchronous:
```
Comment (annnke) [inline docs/design-survey-conditional-logic.md:486]: After you implement an asynchronous function, register it in `FunctionFactory`. The third parameter specifies if this function is asynchronous:

File: docs/design-survey-conditional-logic.md [hunk 6]
```diff
@@ -0,0 +1,691 @@
...
+    ]
+  },
+  // ...
+  ]
+}
+```
+
+[View example](https://surveyjs.io/Examples/Library?id=condition-choicesVisibleIf)
+
+This technique has one drawback: if a question contains many items, you have to copy the same expression into every item that should have dynamic visibility. If that is your case, use a technique described in the next topic.
```
Comment (annnke) [inline docs/design-survey-conditional-logic.md:563]: This technique has one drawback: if a question contains many items, you have to copy the same expression into every item that should have dynamic visibility. If that is your case, use the technique described in the next topic.

File: docs/design-survey-conditional-logic.md [hunk 7]
```diff
@@ -0,0 +1,691 @@
...
+    "setToName": "shippingAddress"
+  }]
+}
+```
+
+---
+
+### `runexpression`
+
+If the `expression` is `true`, runs another expression specified by the `runExpression` property. You can also save the result of `runExpression` as a question value. For this, assign the question's name to the `setToName` property.
```
Comment (annnke) [inline docs/design-survey-conditional-logic.md:689]: If the `expression` is `true`, the trigger runs another expression specified by the `runExpression` property. You can also save the result of `runExpression` as a question value. For this, assign the question's name to the `setToName` property.

**Review by annnke — APPROVED** [review]

**Pull request merged**